src/VisualStudio/VisualBasic/Impl/LanguageService/VisualBasicPackage.vb

```diff
@@ -73,12 +75,9 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic
                           Try
                               RegisterLanguageService(GetType(IVbCompilerService), Function() Task.FromResult(_comAggregate))
 
-                              RegisterService(Of IVbTempPECompilerFactory)(
-                            Async Function(ct)
-                                Dim workspace = Me.ComponentModel.GetService(Of VisualStudioWorkspace)()
-                                Await JoinableTaskFactory.SwitchToMainThreadAsync(ct)
-                                Return New TempPECompilerFactory(workspace)
-                            End Function)
+                              DirectCast(Me, IServiceContainer).AddService(
+                                  GetType(IVbTempPECompilerFactory),
+                                  Function(_1, _2) New TempPECompilerFactory(Me.ComponentModel.GetService(Of VisualStudioWorkspace)()))
                           Catch ex As Exception When FatalError.ReportAndPropagateUnlessCanceled(ex)
                               Throw ExceptionUtilities.Unreachable
                           End Try
```

Comment:
same thing here.  this can all avoid this packageinitializetasks work if not needed.